compile error in pa_win_wasapi.c (cross compile for win32)

**Describe the bug**
I'm trying to cross-compile performous according to [this](https://github.com/retorquere/performous/blob/master/.github/workflows/windows.yml) recipe, and the build ends with the message in the attached log
[message.txt](https://github.com/PortAudio/portaudio/files/5592096/message.txt) (see below)


**To Reproduce**

Run the build workflow

**Expected behavior**

The source tests against WIN64 at pa_win_wasapi.c:171:24. Maybe it should also test against WIN32? I'm not too familiar with Windows alas.

**Actual behavior**

Error messages or logs longer than a page should be attached as a .txt file.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.10, cross-compiling for WIN32
 - OS Version: 20.10
 - PortAudio version: nightly snapshot downloaded from http://www.portaudio.com/archives/pa_snapshot.tgz on 2020-11-24, checksum with `openssl dgst -sha256 pa_snapshot.tgz` is `c020ad3a40ef7dae12dce4ac1b9052b544a9fe43527093b72b78c7e81749ceea`
 - If Windows or Linux, which Host API (e.g. WASAPI): WASAPI from the looks of it.

The error messages are:

```
<command-line>:0:0: note: this is the location of the previous definition
../portaudio/src/hostapi/wasapi/pa_win_wasapi.c:171:24: error: redefinition of 'struct _BYTE_BLOB'
         typedef struct _BYTE_BLOB
                        ^
In file included from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/combaseapi.h:153:0,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/objbase.h:14,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/ole2.h:17,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/wtypes.h:12,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/winscard.h:10,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/windows.h:97,
                 from ../portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
/home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/wtypesbase.h:316:16: note: originally defined here
 typedef struct _BYTE_BLOB {
                ^
../portaudio/src/hostapi/wasapi/pa_win_wasapi.c:175:15: error: conflicting types for 'BYTE_BLOB'
         }     BYTE_BLOB;
               ^
In file included from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/combaseapi.h:153:0,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/objbase.h:14,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/ole2.h:17,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/wtypes.h:12,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/winscard.h:10,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/windows.h:97,
                 from ../portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
/home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/wtypesbase.h:319:3: note: previous declaration of 'BYTE_BLOB' was here
 } BYTE_BLOB;
   ^
../portaudio/src/hostapi/wasapi/pa_win_wasapi.c:176:65: error: conflicting types for 'UP_BYTE_BLOB'
         typedef /* [unique] */  __RPC_unique_pointer BYTE_BLOB *UP_BYTE_BLOB;
                                                                 ^
In file included from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/combaseapi.h:153:0,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/objbase.h:14,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/ole2.h:17,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/wtypes.h:12,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/winscard.h:10,
                 from /home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/windows.h:97,
                 from ../portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
/home/runner/work/performous/performous/win32/performous-mxe/mxe/usr/i686-w64-mingw32.shared/include/wtypesbase.h:321:20: note: previous declaration of 'UP_BYTE_BLOB' was here
 typedef BYTE_BLOB *UP_BYTE_BLOB;
```

Very nice, that works!

I didn't test if the generated library actually runs on Windows (I'm not a Windows user myself), but at least it compiles without error.

BTW, there is a warning about another preprocessor definition:

```
src/hostapi/dsound/pa_win_ds.c:54:0: warning: "_WIN32_WINNT" redefined
 #define _WIN32_WINNT 0x0400 /* required to get waitable timer APIs */
 ^
[...]/mxe/usr/i686-w64-mingw32.static/include/_mingw.h:233:0: note: this is the location of the previous definition
 #define _WIN32_WINNT 0x502
 ^
```

I have no idea whether that matters or not, I just thought I'd mention it while we are at it ...

--- a/src/hostapi/wasapi/pa_win_wasapi.c
+++ b/src/hostapi/wasapi/pa_win_wasapi.c
@@ -168,12 +168,14 @@
         #include <oleidl.h>
         #include <objidl.h>
     #else
-        typedef struct _BYTE_BLOB
-        {
-            unsigned long clSize;
-            unsigned char abData[ 1 ];
-        }     BYTE_BLOB;
-        typedef /* [unique] */  __RPC_unique_pointer BYTE_BLOB *UP_BYTE_BLOB;
+        #ifndef _BLOB_DEFINED
+            typedef struct _BYTE_BLOB
+            {
+                unsigned long clSize;
+                unsigned char abData[ 1 ];
+            }     BYTE_BLOB;
+            typedef /* [unique] */  __RPC_unique_pointer BYTE_BLOB *UP_BYTE_BLOB;
+        #endif
         typedef LONGLONG REFERENCE_TIME;
         #define NONAMELESSUNION
     #endif

